File: MAP/lab2/src/models/VarExp.java
package models;

import utils.IDictionary;

public class VarExp implements IExpression {
    String id;
    public VarExp(String _id){
        id = _id;
    }

    @Override
    public int eval(IDictionary<String, Integer> d){

        return d.getValue(id);
    }

    @Override
    public String toString(){
        return id;
    }

}

File: MAP/lab7/lab7/Models/VarExp.cs
using System;
using lab7.Utils;

namespace lab7.Models
{
    public class VarExp : IExpression
    {
        private String id;

        public VarExp(string id)
        {
            this.id = id;
        }

        public int eval(IDictionary<string, int> dict)
        {
            return dict.getValue(id);
        }

        public override string ToString()
        {
            return id;
        }
    }
}